src/System.Memory/ref/System.Memory.cs

```diff
@@ -77,10 +77,7 @@ public static class Span
         public static ReadOnlySpan<byte> AsBytes<T>(this ReadOnlySpan<T> source) where T : struct { throw null; }
         public static Span<TTo> NonPortableCast<TFrom, TTo>(this Span<TFrom> source) where TFrom : struct where TTo : struct { throw null; }
         public static ReadOnlySpan<TTo> NonPortableCast<TFrom, TTo>(this ReadOnlySpan<TFrom> source) where TFrom : struct where TTo : struct { throw null; }
-
-        public static ReadOnlySpan<char> Slice(this string text) { throw null; }
-        public static ReadOnlySpan<char> Slice(this string text, int start) { throw null; }
-        public static ReadOnlySpan<char> Slice(this string text, int start, int length) { throw null; }
+        public static ReadOnlySpan<char> AsSpan(this string text) { throw null; }
```

Comment:
Since we are keeping NonPortableCast and AsBytes/etc, we can't really remove the Span static class which contains extension methods that have different implementations between portable/fast span.

Is there a way to get around having the extensions methods on Span be split between the two classes?
Currently, we have SpanExtensions for extension methods with implementation solely in corefx while Span has the extension methods for the rest.

Does having the methods in difference static class matter?